src/GeneratesUuid.php

```diff
@@ -121,17 +122,28 @@ public function resolveUuidVersion(): string
      */
     public function scopeWhereUuid($query, $uuid, $uuidColumn = null): Builder
     {
-        $uuidColumn = ! is_null($uuidColumn) && in_array($uuidColumn, $this->uuidColumns())
-            ? $uuidColumn
-            : $this->uuidColumns()[0];
+        $uuidColumn = $this->guessUuidColumn($uuidColumn);
```

Comment:
This refactor improves code reusability by extracting the logic of guessing the UUID column and preparing UUID values into separate methods, `guessUuidColumn` and `prepareUuid`.
These methods are now used in the scopeWhereUuid method and will also be reusable in the upcoming scopeWhereNotUuid method. This reduces redundancy and makes the logic cleaner for handling UUID-related scopes.